internal/iamauth/responses/arn.go

```diff
@@ -0,0 +1,94 @@
+package responses
```

Comment:
I split off this package to avoid an import cycle in the tests.

* I had `iamauth > iamauthtest > iamauth`
* Now, `iamauth > responses` and `iamauth > responsestest` and `iamauth > iamauthtest`

Works, but seems kind of overly nested. I followed what `internal/go-sso` did for test utils, so curious if I'm using `<package>test` the right way, or if I could organize this differently.